src/components/Chat/Chat.tsx

```diff
@@ -309,6 +397,108 @@ const ChatMessage = ({
         >
           <Comment.Text className="light">{!cmd && msg}</Comment.Text>
         </Linkify>
+        <div className={classes.commentMenu}>
+          <Icon
+            onClick={(e: MouseEvent) => {
+              const viewportOffset = (e.target as any).getBoundingClientRect();
+              setReactionMenu(
+                true,
+                id,
+                timestamp,
+                viewportOffset.top,
+                viewportOffset.right
+              );
+            }}
+            name={'' as any}
+            inverted
+            link
+            disabled={isChatDisabled}
+            style={{
+              opacity: 1,
+              display: 'flex',
+              justifyContent: 'center',
+              alignItems: 'center',
+              padding: 10,
+              margin: 0,
+            }}
+          >
+            <span
+              role="img"
+              aria-label="React"
+              style={{ margin: 0, fontSize: 18 }}
+            >
+              😀
+            </span>
+          </Icon>
+        </div>
+        <TransitionGroup>
+          {Object.keys(reactions ?? []).map((key) =>
+            reactions?.[key].length ? (
+              <CSSTransition
+                key={key}
+                timeout={200}
+                classNames={{
+                  enter: classes['reaction-enter'],
+                  enterActive: classes['reaction-enter-active'],
+                  exit: classes['reaction-exit'],
+                  exitActive: classes['reaction-exit-active'],
+                }}
+                unmountOnExit
+              >
+                <Popup
+                  content={reactions[key]
+                    .map((id) => nameMap[id] || 'Unknown')
```

Comment:
we may want to cap this to a certain number of people displayed, if a lot of people react the same way in a large room